05-UI-ReactJS/topic-3/src/components/AddMovie.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+
+class AddMovie extends React.Component {
+  constructor() {
+    super();
+    this.state = { //each component has his own state, so to save i send it to the app state
+      newMovie:{}
+    }
+  }
+  static defaultProps = {
+    categories: ['fiction','comedy','drama']
+  }
+  handleSubmit(event) {
+    if (this.refs.title.value === ''){
+      alert('title required');
+    }else{
+      this.setState({newMovie: {
+        title: this.refs.title.value,
```

Comment:
Is not a good idea to handle inputs with refs. I'd instead use onChange event to trigger state changes.